azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+package azkaban.container.models;
+
+import io.kubernetes.client.custom.Quantity;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1ContainerBuilder;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1EnvVarBuilder;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1ObjectMetaBuilder;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodBuilder;
+import io.kubernetes.client.openapi.models.V1ResourceRequirements;
+import io.kubernetes.client.openapi.models.V1ResourceRequirementsBuilder;
+import io.kubernetes.client.openapi.models.V1Volume;
+import io.kubernetes.client.openapi.models.V1VolumeBuilder;
+import io.kubernetes.client.openapi.models.V1VolumeMount;
+import io.kubernetes.client.openapi.models.V1VolumeMountBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A wrapper Kubernetes Pod builder which internally utilizes
+ * @see io.kubernetes.client.openapi.models to create kubernetes "Pod" resource
+ *
+ * This class provides custom builder methods to have tight coupling of
+ * volumes and init containers (associated with job-types) with application-container
+ * enabling better type safety and compact implementation
+ *
+ * The Pod-spec created using this builder creates one application-container
+ * and multiple init containers corresponding to each jobType
+ */
+public class AzKubernetesV1PodBuilder {
+    private static final String JOBTYPE_VOLUME_PREFIX = "jobtype-volume-";
+    private static final String JOBTYPE_INIT_PREFIX = "jobtype-init-";
+    private static final String AZ_CLUSTER_KEY = "AZ_CLUSTER";
+    private static final String AZ_CONF_VERSION_KEY = "AZ_CONF_VERSION";
+    private static final String JOBTYPE_MOUNT_PATH_KEY = "JOBTYPE_MOUNT_PATH";
+    private static final String DEFAULT_RESTART_POLICY = "Never";
+    private static final String API_VERSION = "v1";
+    private static final String KIND = "Pod";
+
+    private final V1ObjectMetaBuilder podMetaBuilder = new V1ObjectMetaBuilder();
+    private final V1ContainerBuilder flowContainerBuilder = new V1ContainerBuilder();
+
+    private final List<V1VolumeMount> appVolumeMounts = new ArrayList<>();
+    private final List<V1Volume> appVolumes = new ArrayList<>();
+    private final List<V1Container> initContainers = new ArrayList<>();
+
+    private final V1EnvVar azClusterName;
+    private final String restartPolicy;
+
+    /**
+     * @param podName Name of the kubernetes "Pod" resource unique within the namespace
+     * @param podNameSpace Name of the kubernetes "Pod" namespace
+     * @param azClusterName Name to uniquely represent Azkaban instance requesting creation of Pod
+     * @param restartPolicy Optional argument to specify flow container restart policy, otherwise, its "Never"
```

Comment:
What all options are there for restart policy?

Reply:
Always -> All the containers will be restarted if they exit either successfully or with failure
OnFailure -> All the containers which exited with failures will be restarted
Never -> None of the containers will be restarted regardless of their exit status

Restart can be configured with the "maxRetries" property. In our case, application-container/flow-container represents a flow execution. Since we don't have a flow retry policy itself, the default is kept as "Never". However, for future purposes, it may be worth to keep this field.

Reply:
Shall we have enum for it? It will be useful for future reference.